datasetinsights/scripts/public_download.py

```diff
@@ -34,20 +41,37 @@ def parse_args():
         default=const.DEFAULT_DATA_ROOT,
         help="root directory of datasets",
     )
+    parser.add_argument(
+        "--source-uri", default="None", help="URL of synthdet dataset.",
+    )
+
     args = parser.parse_args()
 
     return args
 
 
-def download(name, data_root, version):
+def download(name, source_uri, data_root, version):
     # TODO this method should be refactored once we have clean download
     # CLI interface.
-    dataset = Dataset.find(name)
-    dataset.download(data_root, version)
+    if name.lower() == "Synthetic".lower():
+        SynDetection2D.download(source_uri, data_root)
```

Comment:
Do we need to add source_uri in cli for Synthetic data? In GroceriesReal, we just hard coded the source_uri. It would be better to keep it consistent for both synthetic and real datasets. In that case, we can set `--name SynDetection2D`, and use `dataset = Dataset.find(name)` `dataset.download(source_data_root)`

Reply:
The main reason for adding `source-uri` is that if user wants to use own dataset.

Reply:
But still we would need this because we don't pass version to Synthetic dataset as of now. Can we make synthetic version as `5k`, `10k` etc which are essentially according  to the size of the dataset?

Reply:
Maybe we can name the synthetic dataset version as v1, v2 as well. And write a doc talking about what those versions mean, such as "v1: 5k; v2: 10k". And use a name table to save public uri as well as checksum. Something like this in groceries_real.py:
```
GROCERIES_REAL_DATASET_TABLES = {
        "v3": GroceriesRealTable(
            "v3", "v3.zip", f"{PUBLIC_GROCERIES_REAL_PATH}/v3.zip", 2322380006,
        ),
    }
```